fiaas_skipper/__init__.py

```diff
@@ -34,21 +38,39 @@ def init_k8s_client(config):
     k8s_config.debug = config.debug
 
 
+def _load_spec_config(spec_file):
+    with open(spec_file, 'r') as stream:
+        return yaml.safe_load(stream)
+
+
 def main():
     cfg = Configuration()
     init_logging(cfg)
     init_k8s_client(cfg)
     log = logging.getLogger(__name__)
     try:
-        log.info("fiaas-skipper starting with configuration {!r}".format(cfg))
+        log.info("essgettittt fiaas-skipper starting with configuration {!r}".format(cfg))
         cluster = Cluster()
-        release_channel_factory = ReleaseChannelFactory(cfg.baseurl)
+        if cfg.release_channel_metadata:
+            log.debug("!!Using hardcoded release channel metadata {!r}".format(cfg.release_channel_metadata))
+            release_channel_factory = FakeReleaseChannelFactory(json.loads(cfg.release_channel_metadata))
+        else:
+            release_channel_factory = ReleaseChannelFactory(cfg.baseurl)
+        if os.path.isfile(cfg.spec_file):
+            try:
+                spec_config = _load_spec_config(cfg.spec_file)
+                log.debug("Loaded spec config from file {!r}".format(cfg.spec_file))
+            except yaml.YAMLError:
+                log.exception("Unable to load spec config file {!r} using defaults".format(cfg.spec_file))
+                spec_config = None
```

Comment:
If you set `spec_config=None` before the if-test, you can drop the else-block and this line, making the flow easier to follow.

Reply:
Good point. I'll add that.